### PR TITLE
test: improve backend test coverage to 95.92%

### PR DESCRIPTION
**Pull request by mnbf9rca**
## Summary

Improved backend test coverage from 95.69% to 95.92% while reducing test redundancy and following KISS, DRY, YAGNI principles for a maintainable test suite.

### Coverage Improvement
- **Before:** 95.69% coverage, 1,591 tests
- **After:** 95.92% coverage, 1,592 tests
- **Key improvement:** `app/api/tfl.py` coverage 78.21% → 97.44%

#### 1. Fixture Cleanup
- Removed 3 duplicate `async_client_with_db` fixture definitions from test files
- Added guidance to `AGENTS.md` about using `conftest.py` for shared fixtures

#### 2. Test Consolidation (DRY)
- Removed 5 trivial "task exists" tests that only checked `is not None`
- Parameterized 5 TLS settings tests → 1 test with 6 scenarios
- Parameterized 4 unauthenticated endpoint tests → 1 test with 4 scenarios
- **Net result:** Removed 12 tests, added 5 new tests = 1,592 total

#### 3. Coverage Addition
Added 5 new tests for TfL API metadata endpoints:
- `test_get_severity_codes_success`
- `test_get_disruption_categories_success`
- `test_get_stop_types_success`
- `test_get_line_states_success`
- `test_get_alert_config_success`

### Files Changed
- `AGENTS.md` - Added fixture usage guidance
- `backend/tests/test_admin_alerts.py` - Removed duplicate fixture
- `backend/tests/test_admin_dashboard.py` - Removed duplicate fixture
- `backend/tests/test_admin_route_index.py` - Removed duplicate fixture
- `backend/tests/test_celery_metadata_and_graph_tasks.py` - Removed trivial tests
- `backend/tests/test_celery_tasks.py` - Removed trivial tests
- `backend/tests/test_email_service.py` - Parameterized TLS tests
- `backend/tests/test_tfl_api.py` - Consolidated auth tests + added metadata tests

### Validation
- ✅ All 1,592 tests passing (165.28s runtime)
- ✅ All pre-commit hooks passing (ruff, mypy, prettier, eslint)
- ✅ No linting errors or type checking issues

### Remaining Coverage Gaps (Acceptable)
Per project guidelines, these files remain below 95% and are acceptable:
- `app/cli.py`: 65.14% - CLI tool (low ROI for testing)
- `app/celery/app.py`: 84.09% - Worker startup signals (hard to test)
- `app/celery/database.py`: 89.36% - Worker cleanup paths
- `app/celery/tasks.py`: 88.41% - Retry logic (complex to test)

All other files (46) have 100% coverage or are at target levels (95%+).

## Test Plan
- [x] Run full test suite with coverage: `dotenvx run -- uv run pytest --cov=app`
- [x] Verify all pre-commit hooks pass: `pre-commit run --all-files`
- [x] Confirm test count and coverage improvements

## Summary by Sourcery

Increase backend test coverage for TfL metadata and email TLS handling while consolidating redundant tests and fixtures for improved maintainability.

Enhancements:
- Deduplicate async_client_with_db fixtures across admin-related test modules in favor of shared test configuration.

Documentation:
- Document backend test fixture usage guidelines in AGENTS.md, encouraging reuse via backend/tests/conftest.py.

Tests:
- Add comprehensive tests for TfL metadata and line state endpoints, including severity codes, disruption categories, stop types, line states, and alert config responses.
- Consolidate multiple unauthenticated access tests into a single parameterized authentication test covering several TfL and admin endpoints.
- Parameterize email TLS setting tests across ports and TLS requirements to reduce duplication while preserving behavior coverage.
- Remove trivial Celery task existence tests that only asserted function presence to streamline the test suite.

### CODE REVIEW

File: backend/tests/test_email_service.py [hunk 1]
```diff
@@ -11,26 +11,25 @@
...
+    @pytest.mark.parametrize(
+        ("port", "require_tls", "expected"),
+        [
+            # Port 465 uses implicit TLS regardless of require_tls
+            (465, False, (True, None)),
+            (465, True, (True, None)),
+            # Port 587 with require_tls=False uses auto-upgrade
+            (587, False, (False, None)),
+            # Port 587 with require_tls=True forces STARTTLS
+            (587, True, (False, True)),
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_email_service.py:23]: Partially implemented in commit ae2d1a5ff644fe8cc0e6c93547c045a065037e93.

**Accepted:** Added non-standard port (2525) test cases to document fallback behavior.

**Rejected:** Invalid `require_tls` type tests. The function uses type hints (`require_tls: bool`) and relies on static type checking (mypy) rather than runtime validation. Adding runtime type validation would be scope creep, and testing undefined behavior for wrong types adds no value. This aligns with project principles (KISS, YAGNI).

File: backend/tests/test_tfl_api.py [hunk 2]
```diff
@@ -914,3 +890,179 @@ async def test_get_station_routes_multiple_routes_same_line(
...
+            "description": "Good Service",
+            "last_updated": datetime(2025, 1, 1, tzinfo=UTC),
+        },
+        {
+            "mode_id": "tube",
+            "severity_level": 6,
+            "description": "Severe Delays",
+            "last_updated": datetime(2025, 1, 1, tzinfo=UTC),
+        },
+    ]
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_tfl_api.py:931]: Implemented in commit ae2d1a5ff644fe8cc0e6c93547c045a065037e93. Updated mock to return `SeverityCode` ORM objects to match the existing test pattern used in `test_get_lines_success`.

File: backend/tests/test_tfl_api.py [hunk 3]
```diff
@@ -914,3 +890,179 @@ async def test_get_station_routes_multiple_routes_same_line(
...
+            "category_name": "RealTime",
+            "description": "Real-time disruption",
+            "last_updated": datetime(2025, 1, 1, tzinfo=UTC),
+        },
+        {
+            "category_name": "PlannedWork",
+            "description": "Planned engineering work",
+            "last_updated": datetime(2025, 1, 1, tzinfo=UTC),
+        },
+    ]
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_tfl_api.py:966]: Implemented in commit ae2d1a5ff644fe8cc0e6c93547c045a065037e93. Updated mock to return `DisruptionCategory` ORM objects to match the existing test pattern.

File: backend/tests/test_tfl_api.py [hunk 4]
```diff
@@ -914,3 +890,179 @@ async def test_get_station_routes_multiple_routes_same_line(
...
+        {
+            "type_name": "NaptanMetroStation",
+            "description": "Underground Station",
+            "last_updated": datetime(2025, 1, 1, tzinfo=UTC),
+        },
+        {
+            "type_name": "NaptanRailStation",
+            "description": "Rail Station",
+            "last_updated": datetime(2025, 1, 1, tzinfo=UTC),
+        },
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_tfl_api.py]: Implemented in commit ae2d1a5ff644fe8cc0e6c93547c045a065037e93. Updated mock to return `StopType` ORM objects to match the existing test pattern.

File: backend/tests/test_tfl_api.py [hunk 5]
```diff
@@ -13,6 +13,7 @@
 from app.main import app
 from app.models.tfl import Line, Station
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_tfl_api.py]: Implemented in commit ae2d1a5ff644fe8cc0e6c93547c045a065037e93. Added imports for `SeverityCode`, `DisruptionCategory`, and `StopType` ORM models.

**Comment by mnbf9rca** [comment]
All review comments addressed in commit a85a766.

## Changes Made

### Accepted Suggestions (7 items)

1. ✅ **Added ORM imports** - Imported `SeverityCode`, `DisruptionCategory`, `StopType` models
2. ✅ **Use ORM objects in mocks** - Updated all three metadata tests to use ORM objects instead of dicts
3. ✅ **Extract unauthenticated client fixture** - Created `unauthenticated_client` fixture
4. ✅ **Use method dict** - Refactored `test_endpoints_require_authentication` to use method-to-callable mapping
5. ✅ **Add line-states edge cases** - Added `test_get_line_states_empty` and `test_get_line_states_single_with_no_reason`
6. ✅ **Add non-standard port test** - Added test cases for port 2525 to document fallback behavior

### Partial Acceptance (1 item)

7. ⚠️ **Non-standard port tests** - **Accepted** non-standard port (2525) tests. **Rejected** invalid `require_tls` type tests (relies on static type checking per KISS/YAGNI principles).

## Test Results

- ✅ All 49 tests passing
- ✅ All pre-commit hooks passing
- ✅ No linting errors
- ✅ Type checking passed

File: backend/tests/test_email_service.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_email_service.py:23]: Partially implemented in commit a85a766.

**Accepted:** Added non-standard port (2525) test cases to document fallback behavior.

**Rejected:** Invalid `require_tls` type tests. The function uses type hints (`require_tls: bool`) and relies on static type checking (mypy) rather than runtime validation. Adding runtime type validation would be scope creep, and testing undefined behavior for wrong types adds no value. This aligns with project principles (KISS, YAGNI).

File: backend/tests/test_tfl_api.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_tfl_api.py:931]: Implemented in commit a85a766. Updated mock to return `SeverityCode` ORM objects to match the existing test pattern used in `test_get_lines_success`.

File: backend/tests/test_tfl_api.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_tfl_api.py:966]: Implemented in commit a85a766. Updated mock to return `DisruptionCategory` ORM objects to match the existing test pattern.

File: backend/tests/test_tfl_api.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_tfl_api.py]: Implemented in commit a85a766. Updated mock to return `StopType` ORM objects to match the existing test pattern.

File: backend/tests/test_tfl_api.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_tfl_api.py]: Implemented in commit a85a766. Added imports for `SeverityCode`, `DisruptionCategory`, and `StopType` ORM models.
